src/WebJobs.Script/Config/TransmissionStatusTelemetryModule.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.ApplicationInsights.Extensibility.Implementation;
+using Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Script.Config
+{
+    /// <summary>
+    /// Transmission Status Telemetry Module
+    /// </summary>
+    internal class TransmissionStatusTelemetryModule : ITelemetryModule, IDisposable
```

Comment:
Do we need a new module for this? Can't we just subscribe to the event on TelemetryConfiguration.TelemetryChannel in the config action we're already specifying [here](https://github.com/Azure/azure-functions-host/blob/54941dba8cece0c93744c9bc6a23ea469d0873b7/src/WebJobs.Script/ScriptHostBuilderExtensions.cs#L390)?

Reply:
yes, we can subscribe to the event. I thought telemetry module would be a cleaner approach. Do you see/anticipate any issue with an additional module?

Reply:
Cold start impact? Now one more service needs to be resolved by the DI container.

Reply:
Yes, in addition there'll be a perf impact (however small) of AI having to dispatch to an additional module. If we can do this with a lot less code, I don't know why we wouldn't. The less code the better :) @brettsam may have thoughts on this as well.

Reply:
My only thought is that besides being more natural, the Module gives us a nice Dispose where we can unregister the Handler, which I think we'd need to do. 

I'm not sure what the right approach is if we don't use the Module.

Reply:
I don't see us unregistering the handler in Dispose below? Anyhow, if you don't see any big issues with use of a new module I'm fine with it. I don't have much experience with modules, and was looking for an easier way. If we too the approach I mentioned above and registered the handler, it's just a one liner to subscribe to the event. It's not clear to me whether we need to worry about unsubscribe - what's the lifetime of the telemetry channel? Perhaps when the host is disposed it gets disposed as well.

Reply:
Removed the DI